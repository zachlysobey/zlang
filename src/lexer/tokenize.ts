import { Token } from "../token/token";

export function tokenize(zlang_source: string): Token[] {
  return zlang_source.split("\n").map((s) => ({
    type: "token",
    value: s,
  }));
}
