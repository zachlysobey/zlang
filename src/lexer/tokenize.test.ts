import { file_to_string } from "../test_fixtures/file_to_string";
import { tokenize } from "./tokenize";

describe("tokenize", () => {
  it("needs tests", async () => {
    const input = await file_to_string("assignment.zl");
    const result = tokenize(input);
    expect(result).toMatchSnapshot();
  });
});
