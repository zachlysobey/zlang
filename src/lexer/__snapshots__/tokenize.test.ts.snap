// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tokenize needs tests 1`] = `
Array [
  Object {
    "type": "token",
    "value": "my number",
  },
  Object {
    "type": "token",
    "value": "    3",
  },
  Object {
    "type": "token",
    "value": "",
  },
]
`;
